examples/device/hid_multipleinterface/src/main.c

```diff
@@ -0,0 +1,208 @@
+ * - 1000 ms : device mounted
+ * - 2500 ms : device is suspended
+ */
+enum  {
+  BLINK_NOT_MOUNTED = 250,
+  BLINK_MOUNTED = 1000,
+  BLINK_SUSPENDED = 2500,
+};
+
+static uint32_t blink_interval_ms = BLINK_NOT_MOUNTED;
+
+void led_blinking_task(void);
+void hid_task(void);
+
+/*------------- MAIN -------------*/
+int main(void)
+{
+  board_init();
+  tusb_init();
+
+  while (1)
+  {
+    tud_task(); // tinyusb device task
+    led_blinking_task();
+
+    hid_task();
+  }
+
+  return 0;
+}
+
+//--------------------------------------------------------------------+
+// Device callbacks
+//--------------------------------------------------------------------+
+
+// Invoked when device is mounted
+void tud_mount_cb(void)
+{
+  blink_interval_ms = BLINK_MOUNTED;
+}
+
+// Invoked when device is unmounted
+void tud_umount_cb(void)
+{
+  blink_interval_ms = BLINK_NOT_MOUNTED;
+}
+
+// Invoked when usb bus is suspended
+// remote_wakeup_en : if host allow us  to perform remote wakeup
+// Within 7ms, device must draw an average of current less than 2.5 mA from bus
+void tud_suspend_cb(bool remote_wakeup_en)
+{
+  (void) remote_wakeup_en;
+  blink_interval_ms = BLINK_SUSPENDED;
+}
+
+// Invoked when usb bus is resumed
+void tud_resume_cb(void)
+{
+  blink_interval_ms = BLINK_MOUNTED;
+}
+
+//--------------------------------------------------------------------+
+// USB HID
+//--------------------------------------------------------------------+
+
+void hid_task(void)
+{
+  //Set up interfaces
+  const uint8_t keyboard_interface = 0;
+  const uint8_t mouse_interface = 1;
+
+  // Poll every 10ms
+  const uint32_t interval_ms = 10;
+  static uint32_t start_ms = 0;
+
+  if ( board_millis() - start_ms < interval_ms) return; // not enough time
+  start_ms += interval_ms;
+
+  uint32_t const btn = board_button_read();
+
+  // Remote wakeup
+  if ( tud_suspended() && btn )
+  {
+    // Wake up host if we are in suspend mode
+    // and REMOTE_WAKEUP feature is enabled by host
+    tud_remote_wakeup();
+  }
+
+  /*------------- Keyboard -------------*/
+  if ( tud_hid_n_ready(keyboard_interface) )
+  {
+    // use to avoid send multiple consecutive zero report for keyboard
+    static bool has_key = false;
+
+    if ( btn )
+    {
+      uint8_t keycode[6] = { 0 };
+      keycode[0] = HID_KEY_A;
+
+      tud_hid_n_keyboard_report(keyboard_interface, 0, 0, keycode);
+
+      has_key = true;
+    }else
+    {
+      // send empty key report if previously has key pressed
+      if (has_key) tud_hid_n_keyboard_report(keyboard_interface, 0, 0, NULL);
+      has_key = false;
+    }
+  }
+
+  /*------------- Mouse -------------*/
+  if ( tud_hid_n_ready(mouse_interface) )
+  {
+    if ( btn )
+    {
+      int8_t const delta = 5;
+
+      // no button, right + down, no scroll pan
+      tud_hid_n_mouse_report(mouse_interface, 0, 0x00, delta, delta, 0, 0);
+
+      // delay a bit before attempt to send keyboard report
+      board_delay(10);
```

Comment:
There is no need to delay since they are on separated interfaces. Previously itf is shared, so we have to wait before sending another one.